### ISSUE TITLE
Domain of sender address root@fail2ban does not exist

### ISSUE THREAD

**Issue by JonasAlfredsson** [JonasAlfredsson/docker-fail2ban#2]
I noticed that I no longer get mail sent to me when bans are handed out, so I looked a bit deeper into it. 

Creating a toy example I want to ban the IP `192.168.0.1` after three attempts, and when that happens the output is as follows:

```
2020-03-16 19:52:13,547 fail2ban.actions        [1]: NOTICE  [nextcloud] Ban 192.168.0.1
2020-03-16 19:52:13,998 fail2ban.utils          [1]: #39-Lev. 7fa2c3bf7030 -- exec: printf %b "Subject: [Fail2Ban] nextcloud: banned 192.168.0.1 from silent-pc
Date: `LC_ALL=C date +"%a, %d %h %Y %T %z"`
From: Fail2Ban <root@fail2ban>
To: root@localhost\n
Hi,\n
The IP 192.168.0.1 has just been banned by Fail2Ban after
3 attempts against nextcloud.\n\n
Here is more information about 192.168.0.1 :\n
`/usr/bin/whois 192.168.0.1 || echo missing whois program`\n
Regards,\n
Fail2Ban" | /usr/sbin/sendmail -f root@fail2ban root@localhost
2020-03-16 19:52:13,999 fail2ban.utils          [1]: ERROR   7fa2c3bf7030 -- stderr: 'sendmail: RCPT TO:<root@localhost> (553 5.1.8 <root@localhost>... Domain of sender address root@fail2ban does not exist)'
2020-03-16 19:52:13,999 fail2ban.utils          [1]: ERROR   7fa2c3bf7030 -- returned 1
2020-03-16 19:52:14,000 fail2ban.actions        [1]: ERROR   Failed to execute ban jail 'nextcloud' action 'sendmail-whois' info 'ActionInfo({'ip': '192.168.0.1', 'family': 'inet4', 'fid': <function Actions.ActionInfo.<lambda> at 0x7fa2c0b53ea0>, 'failures': 3, 'F-*': {'matches': [('', '2020-03-16T18:51:51Z', " Random text. Login failed: 'whatever' (Remote IP: '192.168.0.1') More text"), "2020-03-16T18:52:11Z Random text. Login failed: 'whatever' (Remote IP: '192.168.0.1') More text", "2020-03-16T18:52:12Z Random text. Login failed: 'whatever' (Remote IP: '192.168.0.1') More text"], 'failures': 3, 'ip4': '192.168.0.1'}, 'raw-ticket': <function Actions.ActionInfo.<lambda> at 0x7fa2c0b56488>})': Error banning 192.168.0.1
```

The IP is actually successfully banned, as can be seen in the `iptables` output below:

```
Chain f2b-nextcloud (1 references)
target     prot opt source               destination
REJECT     all  --  192.168.0.1          0.0.0.0/0            reject-with icmp-port-unreachable
RETURN     all  --  0.0.0.0/0            0.0.0.0/0
```

However, no mails are sent, and the reason seems to be because I thought it would be useful to default to having the senders address be `root@fail2ban`. The `sendmail` program will reject the sender with this domain, since it can't find proper DNS records of it, and thus figures it will not be possible to send a response back to this sender. 

The solution is to use the "hostname" instead, since this will actually be resolvable.

**Comment by JonasAlfredsson** [comment]
Explanation: https://stackoverflow.com/a/17742092

**Comment by JonasAlfredsson** [comment]
Solved in commit [4baf1d3](https://github.com/JonasAlfredsson/docker-fail2ban/commit/4baf1d36fa7eb239ac1f7548b657f1ac4dd12e89).
